app/src/main/java/com/hedvig/app/service/badge/NotificationBadgeService.kt

```diff
@@ -0,0 +1,50 @@
+package com.hedvig.app.service.badge
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+interface NotificationBadgeService {
```

Comment:
I'm personally not a big fan of single-implementation interfaces - I feel like they generally just add a layer of abstraction that never actually delivers any concrete value.

The way I see it, in Java/C#, there was an 'acceptable reason' to create them in that mocking classes was far easier with single implementation-interfaces when testing, but `mockk` can mock any given non-open class just fine, making that reason kind of moot.

Some more reading on the subject: https://www.symphonious.net/2011/06/18/the-single-implementation-fallacy/ and https://martinfowler.com/bliki/InterfaceImplementationPair.html